src/OpenSearch/Handlers/SigV4Handler.php

```diff
@@ -46,11 +49,15 @@ public function __construct(
     public function __invoke(array $request)
     {
         $creds = call_user_func($this->credentialProvider)->wait();
+
+        // remove body before signing request
```

Comment:
This is a workaround for serverless, but the server is going to fix this AFAIK, shortly. Split up the PR into setting the service name and the workaround for now, and let's not merge the workaround.